CODING_GUIDELINES.md

```diff
@@ -211,44 +217,6 @@ a += "b"
   + "d";
 ```
 
-* Initializers
-
-We allow brace enclosed initializers only for aggregates and arrays/containers.<br />
-Brace enclosed initializer MUST be used with equality sign if it follows the variable declaration.<br />
-Brace enclosed initializer MUST be additionally enclosed in parentheses if it is used in constructor initialization list.<br />
-Some valid use cases:
-```c++
-// aggregate
-Person john = { "John", "Smith", 21 };
-Person *john = new Person { "John", "Smith", 21 };
-
-// array
-int array[] = { 1, 2, 3, 4 };
-
-// container
-QHash<QString, QString> map = {
-    { "key1", "value1" },
-    { "key2", "value2" }
-);
-
-// member array
-SomeClass::SomeClass(BaseClass *parent)
-    : BaseClass(parent)
-    , m_someArrayMember({ 1, 2, 3, 4 })
-{
-}
-
-// return from function
-Person getPersonByName(const QString &name)
-{
-    // do something
-    return { name, surname, age };
-}
-
-// function argument
-doSomething({ name, surname, age }, someOtherArg);
-```
-
 * **auto** keyword
 
 We allow the use of the **auto** keyword only where it doesn't break the readability of the code (i.e. either we can gather enough information about the type from the right part of the expression, or we do not need to know the exact type), or where it is strictly necessary (for example, to compute the type of a lambda, etc.).<br />
```

Comment:
> I don't think he posted a specific rewording.

I did, but it was a quick-written snippet and this a good opportunity to correct mistakes in it. 

We allow the use of the **auto** keyword only  where it is strictly necessary (for example, to declare a lambda object, etc.), or where it **enhances** the readability of the code. Declarations for which one can gather enough information about the object interface (type) from its name or the usage pattern (an iterator or a loop variable are good examples of clear patterns) or the right part of the expression nicely fit here.

When weighing whether to use an auto-typed variable please think about potentials reviewers of your code, who will read it as a plain diff (on github.com, for instance). Please make sure that such reviewers can understand the code completely and without excessive efforts.

Some valid use cases: